From 2876b7fc77ae32a6483621a754d59b4e36616ff5 Mon Sep 17 00:00:00 2001
From: Pierluigi Passaro <pierluigi.p@variscite.com>
Date: Sun, 17 Mar 2019 11:40:53 +0100
Subject: [PATCH 1/5] sysfs sepolicy: allow full access capabilities

Managing GPIOs using sysfs requires full access to:
- write
- read
- create folders (when exporting GPIO id)
- remove folders (when unexporting GPIO id)
---
 prebuilts/api/28.0/public/domain.te | 1 -
 public/domain.te                    | 1 -
 2 files changed, 2 deletions(-)

diff --git a/prebuilts/api/28.0/public/domain.te b/prebuilts/api/28.0/public/domain.te
index e9337b6..d74c25d 100644
--- a/prebuilts/api/28.0/public/domain.te
+++ b/prebuilts/api/28.0/public/domain.te
@@ -1373,7 +1373,6 @@ neverallow { domain -traced_probes } self:capability dac_read_search;
 # to do so should never be granted.
 neverallow domain {
   proc_type
-  sysfs_type
 }:dir { add_name create link remove_name rename reparent rmdir write };
 
 # cgroupfs directories can be created, but not files within them.
diff --git a/public/domain.te b/public/domain.te
index f348701..4adc8cc 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -1344,7 +1344,6 @@ full_treble_only(`
 # to do so should never be granted.
 neverallow domain {
   proc_type
-  sysfs_type
 }:dir { add_name create link remove_name rename reparent rmdir write };
 
 # cgroupfs directories can be created, but not files within them.
-- 
2.25.1

